pkg/util/testing/core.go

```diff
@@ -69,6 +70,34 @@ func CheckLatestEvent(ctx context.Context, k8sClient client.Client,
 	return false, fmt.Errorf("mismatch with the latest event: got r:%v t:%v n:%v, reg %v", item.Reason, item.Type, item.Note, item.Regarding)
 }
 
+// CheckEventRecordedAfter checks if an event identified by eventReason, eventType, eventNote was recorded after the provided time t
+func CheckEventRecordedAfter(ctx context.Context, k8sClient client.Client,
+	eventReason string,
+	eventType string, eventNote string, t time.Time) (bool, error) {
+	events := &eventsv1.EventList{}
+	if err := k8sClient.List(ctx, events, &client.ListOptions{}); err != nil {
+		return false, err
+	}
+
+	length := len(events.Items)
+	if length == 0 {
+		return false, errors.New("no events currently exist")
+	}
+
+	lastIndexAfterTime := 0
```

Comment:
I don't like the optimization for the iteration over the list by timestamp. It adds complexity (I'm not sure we are even guaranteed that the events are sorted). Also, it does not improve performance in any meaningful way, because the full list needs to be fetched from API server (so serialized there), and deserialized in the client. Investigating potential flakes will eat us much more time than the potential gain.

Reply:
reworked